From c3bb69d2c3df365c7b95cd8d519532a2f03e74c0 Mon Sep 17 00:00:00 2001
From: Rastrisr <rastrisr@tutamail.com>
Date: Thu, 15 Oct 2026 02:01:48 +0000
Subject: [PATCH] LineageSettings: add Secure.MAXIMUM_FAILED_PASSWORDS_FOR_WIPE

Needed by CalyxOS wipe-after-failed-unlocks (fw/b 81661fe0, Settings 04e55f1e).

Change-Id: I1006e7f4c7c031c12ce75307579fa2b97421a764
---
 .../java/lineageos/providers/LineageSettings.java  | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/sdk/src/java/lineageos/providers/LineageSettings.java b/sdk/src/java/lineageos/providers/LineageSettings.java
index 2a845bc..4c0dbfc 100644
--- a/sdk/src/java/lineageos/providers/LineageSettings.java
+++ b/sdk/src/java/lineageos/providers/LineageSettings.java
@@ -2961,6 +2961,18 @@ public final class LineageSettings {
         public static final Validator VOLUME_PANEL_ON_LEFT_VALIDATOR =
                 sBooleanValidator;
 
+        /**
+         * Number of failed unlock attempts after which the device (or this user) is wiped,
+         * when no device admin policy sets a stricter value. 0 = never. (from CalyxOS)
+         * @hide
+         */
+        public static final String MAXIMUM_FAILED_PASSWORDS_FOR_WIPE =
+                "maximum_failed_passwords_for_wipe";
+
+        /** @hide */
+        public static final Validator MAXIMUM_FAILED_PASSWORDS_FOR_WIPE_VALIDATOR =
+                new InclusiveIntegerRangeValidator(0, 30);
+
         /**
          * Whether to use black theme for dark mode
          * @hide
@@ -2999,6 +3011,8 @@ public final class LineageSettings {
             VALIDATORS.put(NETWORK_TRAFFIC_SHOW_UNITS, NETWORK_TRAFFIC_SHOW_UNITS_VALIDATOR);
             VALIDATORS.put(TRUST_WARNINGS, TRUST_WARNINGS_VALIDATOR);
             VALIDATORS.put(VOLUME_PANEL_ON_LEFT, VOLUME_PANEL_ON_LEFT_VALIDATOR);
+            VALIDATORS.put(MAXIMUM_FAILED_PASSWORDS_FOR_WIPE,
+                    MAXIMUM_FAILED_PASSWORDS_FOR_WIPE_VALIDATOR);
         }
     }
 
-- 
2.43.0

